### PR TITLE
Initial visionOS integration

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This PR represents a *very* WIP implementation of the FluentUI library and demo app for visionOS. Most components are still pretty broken and not yet supported, but a variety of core scenarios are now viable for additional exploration.

### This is *not* full visionOS support!

That said, this at least unblocks any downstream apps that want to build against visionOS and utilize some aspects of the FluentUI library.

Some important caveats:
* CocoaPods is not yet supported. Only SPM and direct library integration are currently available. We'll continue to monitor visionOS support in CocoaPods and evaluate accordingly.
* Some alias tokens have been updated with hardcoded values. This will be eventually removed and migrated to a proper per-platform override system.
* As mentioned above, many components are either unsupported or have not been throughly verified on visionOS. Please treat this as an experimental WIP.

### Binary change

Note: the script for processing binary changes seems to be having problems with my build. While I investigate, here's the quick breakdown:

| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 31,037,192 bytes | 31,114,936 bytes | ⚠️ 77,744 bytes |

### Verification

iOS app builds and runs as expected. visionOS app detailed below:

<details>
<summary>Visual Verification</summary>

vision1_filtered
vision2
vision3
vision4

</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/1956)

### CODE REVIEW

**Review by lcapkovic — APPROVED** [review]

**Comment by huwilkes** [comment]
Should this go into a separate VisionOS branch for a lot of the same reasons we decided to not let the Catalyst changes into main?

File: ios/xcode/FluentUITests.xcconfig [hunk 1]
```diff
@@ -10,5 +10,7 @@ INFOPLIST_FILE = FluentUI.Tests/Info.plist
 LD_RUNPATH_SEARCH_PATHS = $(inherited) @executable_path/Frameworks @loader_path/Frameworks
 PRODUCT_BUNDLE_IDENTIFIER = com.microsoft.FluentUITests
 PRODUCT_NAME = $(TARGET_NAME)
+SUPPORTED_PLATFORMS = iphoneos iphonesimulator xros xrsimulator
+SUPPORTS_XR_DESIGNED_FOR_IPHONE_IPAD = NO
```
Comment (anandrajeswaran) [inline ios/xcode/FluentUITests.xcconfig:14]: Curious - is this needed? What's the default value?

**Review by joannaquu — APPROVED** [review]

File: ios/xcode/FluentUITests.xcconfig (hunk 1 above)
Comment (Saadnajmi) [reply ios/xcode/FluentUITests.xcconfig:14]: Xcode adds this based on what you select in the GUI if you're creating a new scheme. Not sure if they're needed.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewHeaderFooterViewDemoController.swift [hunk 2]
```diff
@@ -133,9 +133,11 @@ extension TableViewHeaderFooterViewDemoController {
                 }
                 footer?.setup(style: .footer, attributedTitle: title)
 
+#if os(iOS)
```
Comment (Saadnajmi) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewHeaderFooterViewDemoController.swift:136]: Is this a Swift specific ifdef? can I use it in objective C?

Comment (Saadnajmi) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewHeaderFooterViewDemoController.swift:136]: It appears so https://developer.apple.com/documentation/xcode/running-code-on-a-specific-version/

File: ios/xcode/FluentUITests.xcconfig (hunk 1 above)
Comment (mischreiber) [reply ios/xcode/FluentUITests.xcconfig:14]: If you don't include this, the Run Destinations dropdown includes an extra "Apple Vision Pro (Designed for iPad)" entry that we don't want:

VisionProiPad

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewHeaderFooterViewDemoController.swift (hunk 2 above)
Comment (mischreiber) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/TableViewHeaderFooterViewDemoController.swift:136]: Yep it's available in both! For anyone else seeing this:
```swift
#if os(iOS) // or os(visionOS)
// ...
#endif
```

```c
// Objective-C
#if TARGET_OS_IOS // or TARGET_OS_VISION
// ...
#endif
```

**Comment by mischreiber** [comment]
> Should this go into a separate visionOS branch for a lot of the same reasons we decided to not let the Catalyst changes into main?
> 
> Edit:
> 
> * [Fix no such module 'FluentUI_ios' on MacCatalyst #1878](https://github.com/microsoft/fluentui-apple/pull/1878)
> 
> for context

Added clearer indications in the demo as to which controls are enabled. I'll also be working to fix accessibility for a handful of cases where the controls are usable but the demo itself is not to our standards.

**Pull request merged**